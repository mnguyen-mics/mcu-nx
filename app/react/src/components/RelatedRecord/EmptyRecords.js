import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'antd';
import McsIcons from '../McsIcons';

function EmptyRecords({ iconType, message, className }) {
  return (
    <Row className={`empty-related-records ${className}`}>
      <Col span={24}>
        <McsIcons type={iconType} />
      </Col>
      <Col span={24}>
        {message}
      </Col>
    </Row>
  );
}

EmptyRecords.defaultProps = {
  iconType: 'warning',
  className: '',
};

EmptyRecords.propTypes = {
  iconType: PropTypes.string,
  message: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default EmptyRecords;
