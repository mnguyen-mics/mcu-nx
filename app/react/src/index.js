import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
