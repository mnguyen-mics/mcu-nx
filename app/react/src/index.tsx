/* eslint-disable */
import React from 'react';
import { render } from 'react-dom';

import App from './App';

declare global {
  interface Window {
    require: any;
    mainRev: any;
  }
}

render(<App />, document.getElementById('mcs-react-app'), () => {
  window.require([window.mainRev], () => {
    //
  });
});
