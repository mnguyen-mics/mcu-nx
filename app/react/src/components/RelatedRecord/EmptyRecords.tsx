import * as React from 'react';
import McsIcons from '../McsIcons';

export interface EmptyRecordsProps {
  iconType: string;
  message: string;
  className: string;
}

const EmptyRecords: React.SFC<EmptyRecordsProps> = props => {
  return (
    <div className={`empty-related-records ${props.className}`}>
      <McsIcons type={props.iconType} />
      {props.message}
    </div>
  );
};

EmptyRecords.defaultProps = {
  iconType: 'exclamation-circle-o',
  className: '',
};

export default EmptyRecords;
