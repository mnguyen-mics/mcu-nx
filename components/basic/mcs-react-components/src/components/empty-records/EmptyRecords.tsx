import * as React from 'react';
import McsIcon, { McsIconType } from '../mcs-icon';

export interface EmptyRecordsProps {
  iconType?: McsIconType;
  genericIconProps?: JSX.Element;
  message: string;
  className?: string;
}

const EmptyRecords: React.SFC<EmptyRecordsProps> = props => {
  return (
    <div className={`mcs_emptyRelatedRecords ${props.className ? props.className : ''}`}>
      {props.genericIconProps ? props.genericIconProps : <McsIcon type={props.iconType!} />}
      {props.message}
    </div>
  );
};

EmptyRecords.defaultProps = {
  iconType: 'warning',
};

export default EmptyRecords;
