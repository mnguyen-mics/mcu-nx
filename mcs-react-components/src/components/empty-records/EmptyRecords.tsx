
import * as React from 'react';
import McsIcon, { McsIconType } from '../mcs-icon';

export interface EmptyRecordsProps {
  iconType?: McsIconType;
  message: string;
  className?: string;
}

const EmptyRecords: React.SFC<EmptyRecordsProps> = props => {
  return (
    <div className={`mcs_emptyRelatedRecords ${props.className ? props.className : ''}`}>
      <McsIcon type={props.iconType!} />
      {props.message}
    </div>
  );
};

EmptyRecords.defaultProps = {
  iconType: 'warning'
};

export default EmptyRecords;