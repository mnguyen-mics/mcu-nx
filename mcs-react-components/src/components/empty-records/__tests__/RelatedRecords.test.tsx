import 'jest';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import EmptyRecords, { EmptyRecordsProps } from '../EmptyRecords';
import { MemoryRouter } from 'react-router';

it('renders the EmptyRecords', () => {
  const props: EmptyRecordsProps = {
    iconType: 'question',
    message: 'Hello world!',
    className: 'mcs-customClass'  
  };

  const component = TestRenderer.create(
    <MemoryRouter>
      <EmptyRecords {...props} />
    </MemoryRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
