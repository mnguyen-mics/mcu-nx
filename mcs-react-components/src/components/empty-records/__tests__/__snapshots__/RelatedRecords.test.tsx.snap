// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the EmptyRecords 1`] = `
<div
  className="mcs_emptyRelatedRecords mcs-customClass"
>
  <span
    className="mcs-icon "
  >
    <i
      className="mcs-question"
      style={Object {}}
    />
  </span>
  Hello world!
</div>
`;
